lnst/Controller/ContainerPoolManager.py

```diff
@@ -329,3 +368,15 @@ def cleanup_networks(self):
     def cleanup(self):
         self.cleanup_containers()
         self.cleanup_networks()
+        if check_process_running("firewalld"):
```

Comment:
at this point i think you don't have to check that the process is running but that you have done and saved some firewalld configuration in the `self._firewalld_config` dictionary

Reply:
i say that because i think you can change the firewalld configuration with the `firewall-cmd` commands even if the process isn't actively running but i may be misremembering something... probably good to check that, but i don't have firewalld on my laptop.

Reply:
Yeah, I was thinking about the same, what you propose may be better. I will change that.

Reply:
> i say that because i think you can change the firewalld configuration with the firewall-cmd commands even if the process isn't actively running but i may be misremembering something...

I don't understand. If firewalld is not running I assume that the service is not enabled and therefore no need to modify firewall configuration.

Reply:
what i was trying to say is that i seem to remember that firewalld works as a database that is also editable offline when the actual firewall service isn't running - so the check process would return `False` and we'd keep our configuration in the database without cleaning it up.

this would be an edge case of when the service crashes or is somehow shut down during the test which is of course unlikely.

But, my main reason for the proposal is that it also simplifies the code a bit - you don't have to check for a running process, since the database is *always* editable (even if the service is offline), you can simply trust your internal `_firewalld_config` and clean up the database without the process check.

I checked and my assumption about the offline database is true, see:
* https://firewalld.org/documentation/configuration/runtime-versus-permanent.html
* https://firewalld.org/documentation/utilities/firewall-offline-cmd.html

But from the `firewalld-cmd` commands that you use here I think we're only changing the runtime configuration, not the permanent one, which means that in case the service crashes, our configuration automatically disappears.

Still the code is probably a bit simpler without the process check for the cleanup.

Reply:
Ok, let me try this on Fedora and if it works, I'll remove check for firewalld process running.

Reply:
Here's a try with the firewalld check removed:
```
2023-01-25 18:25:42       (localhost)        -   DEBUG: Adding interface cni-podman1 to firewalld zone internal
FirewallD is not running
Command 'firewall-cmd --zone=internal --change-interface=cni-podman1' returned non-zero exit status 252.
```

Not sure if `--permanent` would make any difference, but I'd rather avoid changing permanent configuration.

Eventually, we could check the `firewall-cmd` error code, where the exit code of 252 means the firewalld service is not running. I think this would be worth to try.

Reply:
After offline discussion here's the plan:
* checking whether firewalld is running is better, since if it's running that means the firewall-cmd is available and I don't have to check for its presence
* when check for the return code is used, I'd have to also check whether the fail is caused by firewall-cmd not being present

Additionally, I should double check whether the overall solution is valid with Eric Garver.

One more note, by default the firewalld configuration should not be touched, this should be configurable through ContainerPoolManager setting.